Content.Server/Shuttles/Systems/ShuttleTimerSystem.cs

```diff
@@ -0,0 +1,196 @@
+        }
+
+        /// <summary>
+        /// Update all child screens of parent <see cref="ShuttleTimerComponent"/>.
+        /// </summary>
+        private void UpdateLocalScreens(ShuttleTimerComponent component, TimeSpan duration)
+        {
+            foreach (var timerUid in component.LocalScreens)
+            {
+                RemComp<ActiveTextScreenTimerComponent>(timerUid);
+
+                // this kept not triggering on TimeSpan.Zero
+                if (duration <= TimeSpan.MinValue)
+                    continue;
+
+                // set delay and start
+                var ev = new TextScreenTimerEvent(duration);
+                RaiseLocalEvent(timerUid, ref ev);
+            }
+        }
+
+        /// <summary>
+        /// Update all synced remote screens in a <see cref="ShuttleTimerComponent"/>. Filter by delegate.
+        /// </summary>
+        /// <param name="filter">Delegate determining remote timer update eligibility from shuttleUid's map</param>
+        private void UpdateRemoteScreens(ShuttleTimerComponent component, TimeSpan duration, MapFilter filter)
+        {
+            foreach (var timerUid in component.RemoteScreens)
+            {
+                if (!TryComp<TransformComponent>(timerUid, out var timerXform) || timerXform.MapUid == null)
+                    continue;
+
+                if (!filter.Comparator(timerXform.MapUid.Value, filter.MapUid))
+                    continue;
+
+                RemComp<ActiveTextScreenTimerComponent>(timerUid);
+
+                if (duration <= TimeSpan.MinValue)
+                    continue;
+
+                var ev = new TextScreenTimerEvent(duration);
+                RaiseLocalEvent(timerUid, ref ev);
+            }
+        }
+
+        /// <summary>
+        /// Update all synced remote screens in a <see cref="ShuttleTimerComponent"/>. No filtering.
+        /// </summary>
+        private void UpdateRemoteScreens(ShuttleTimerComponent component, TimeSpan duration)
+        {
+            foreach (var timerUid in component.RemoteScreens)
+            {
+                RemComp<ActiveTextScreenTimerComponent>(timerUid);
+
+                if (duration <= TimeSpan.MinValue)
+                    continue;
+
+                var ev = new TextScreenTimerEvent(duration);
+                RaiseLocalEvent(timerUid, ref ev);
+            }
+        }
+
+        // never got an answer figuring out how to do this. there's definitely a better pattern
+        /// <summary>
+        /// Raises a TEvent on each entity containing TComp.
+        /// </summary>
+        public void RaiseEventOnShuttles<TComp, TEvent>(ref TEvent ev)
+            where TComp : Component
+            where TEvent : struct
+        {
+            var query = AllEntityQuery<TComp>();
+            while (query.MoveNext(out var uid, out _))
+            {
+                RaiseLocalEvent(uid, ref ev);
+            }
+        }
```

Comment:
Just add it to engine, also I don't think the event should have a struct constraint.